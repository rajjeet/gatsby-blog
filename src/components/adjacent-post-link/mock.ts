export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
