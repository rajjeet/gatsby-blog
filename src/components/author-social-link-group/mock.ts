import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

