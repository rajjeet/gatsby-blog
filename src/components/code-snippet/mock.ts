import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
