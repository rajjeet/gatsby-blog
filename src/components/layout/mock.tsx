import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

