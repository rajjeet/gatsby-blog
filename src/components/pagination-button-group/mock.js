export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
