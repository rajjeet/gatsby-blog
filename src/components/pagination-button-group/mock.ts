export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
