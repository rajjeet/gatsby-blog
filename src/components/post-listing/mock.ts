import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

