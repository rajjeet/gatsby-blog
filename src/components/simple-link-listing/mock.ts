import { TProps } from './types';
import { mockProjectLinks } from '../../fixtures/projects';

export const makeProps = (): TProps => ({
  links: mockProjectLinks,
});

