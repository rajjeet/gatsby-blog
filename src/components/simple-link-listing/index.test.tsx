import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { SimpleLinkListing } from './index';
import { makeProps } from './mock';

afterEach(cleanup);

describe('<SimpleLinkListing />', () => {
  it('should render', () => {
    const { asFragment } = render(<SimpleLinkListing {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should has a heading', () => {
    const { getByText } = render(<SimpleLinkListing {...makeProps()} />);
    expect(getByText('Links')).toBeDefined();
  });

  it('should have a list of links', () => {
    const { getByLabelText } = render(<SimpleLinkListing {...makeProps()} />);
    expect(getByLabelText('ex. CharismaCondos.net')).toBeDefined();
    expect(getByLabelText('ex. SouthWoodbridge.com')).toBeDefined();
    expect(getByLabelText('Github Code')).toBeDefined();
  });

  it('should have 3 links', () => {
    const { queryAllByRole } = render(<SimpleLinkListing {...makeProps()} />);
    expect(queryAllByRole('link')).toHaveLength(3);
  });
});
