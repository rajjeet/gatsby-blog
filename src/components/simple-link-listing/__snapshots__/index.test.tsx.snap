// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleLinkListing /> should render 1`] = `
<DocumentFragment>
  .c0 {
  list-style: none;
  margin-left: 0;
  padding-left: 0;
}

.c0 li {
  margin-bottom: .25em;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  padding: .7em;
  border-radius: 5px;
}

.c0 li:hover {
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
  -webkit-transition: ease .3s;
  transition: ease .3s;
}

.c0 a {
  color: black;
}

<div>
    <h3>
      Links
    </h3>
    <ul
      class="c0"
    >
      <a
        aria-label="ex. CharismaCondos.net"
        href="https://dgftd2qqywjy1.cloudfront.net"
        rel="noopener noreferrer"
        target="_blank"
      >
        <li>
          ex. CharismaCondos.net
        </li>
      </a>
      <a
        aria-label="ex. SouthWoodbridge.com"
        href="https://d1n2vqbqyes277.cloudfront.net"
        rel="noopener noreferrer"
        target="_blank"
      >
        <li>
          ex. SouthWoodbridge.com
        </li>
      </a>
      <a
        aria-label="Github Code"
        href="https://github.com/rajjeet/charismacondos"
        rel="noopener noreferrer"
        target="_blank"
      >
        <li>
          Github Code
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
